pages/_document.js

```diff
@@ -2,22 +2,52 @@ import Document, {Head, Main, NextScript} from 'next/document';
 import React from 'react';
 import {ServerStyleSheet} from 'styled-components';
 
-export default class CustomDocument extends Document {
-    static getInitialProps({renderPage}) {
+const GA_TRACKING_ID = 'UA-27106984-1';
+
+class CustomDocument extends Document {
+    static async getInitialProps(ctx) {
         const sheet = new ServerStyleSheet();
-        const page = renderPage((App) => (props) => sheet.collectStyles(<App {...props} />));
-        const styleTags = sheet.getStyleElement();
+        const originalRenderPage = ctx.renderPage;
+
+        try {
+            // eslint-disable-next-line no-param-reassign
+            ctx.renderPage = () =>
```

Comment:
This is the new hotness from the most updated styled-components example. Prevents memory leaks, IIRC.

Reply:
😍

Reply:
Even more hotness: plugins. Won't land in master for while, tho. 

https://github.com/explodingcamera/next-plugin-styled-components/tree/9086ef49af49af83b1ac70ff59c1b4053b84eb65/packages/next-plugin-styled-components/src